### ISSUE TITLE
[BUG] Error setting network-mode on non linux platforms

### ISSUE THREAD

**Issue by adrianriobo** [crc-org/crc#3796]
Recently a `story_microshift` was added to the e2e scenarios by #3779 , the background steps include setting the network-mode to user.

On non-linux platforms that config option does not exist as network-mode user is the only mode available; this results on an error executing the story on non-linux platforms:

```
<testsuites name="crc" tests="1" skipped="0" failures="1" errors="0" time="0.6364514">
  <testsuite name="Microshift test stories" tests="1" skipped="0" failures="1" errors="0" time="0.6312547">
    <testcase name="Start and expose a basic HTTP service and check after restart" status="failed" time="0.6312547">
      <failure message="Step setting config property &#34;network-mode&#34; to value &#34;user&#34; succeeds: command &#39;crc config set network-mode user&#39;, expected to succeed, exited with exit code: 1&#xA;Command stdout: &#xA;Command stderr: level=debug msg=&#34;CRC version: 2.25.0+97705a\n&#34;&#xA;level=debug msg=&#34;OpenShift version: 4.13.6\n&#34;&#xA;level=debug msg=&#34;Podman version: 4.4.4\n&#34;&#xA;level=debug msg=&#34;Running &#39;crc config set&#39;&#34;&#xA;Configuration property &#39;network-mode&#39; does not exist&#xA;"></failure>
...
```

**Comment by praveenkumar** [comment]
@adrianriobo I tested the scenario in mac and it does have `network-mode` setting. Is it only for windows?

```
✗ uname -a
Darwin prkumar-mac 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 arm64

$ PULL_SECRET_FILE=/Users/prkumar/pull-secret BUNDLE_LOCATION=/Users/prkumar/.crc/cache/crc_microshift_vfkit_4.13.6_arm64.crcbundle make e2e-story-microshift 
go install -tags "containers_image_openpgp"  -ldflags="-X github.com/crc-org/crc/pkg/crc/version.crcVersion=2.25.0 -X github.com/crc-org/crc/pkg/crc/version.ocpVersion=4.13.6 -X github.com/crc-org/crc/pkg/crc/version.okdVersion=4.13.0-0.okd-2023-06-04-080300 -X github.com/crc-org/crc/pkg/crc/version.podmanVersion=4.4.4 -X github.com/crc-org/crc/pkg/crc/version.microshiftVersion=4.13.6 -X github.com/crc-org/crc/pkg/crc/version.commitSha=97705a "  ./cmd/crc
Log successfully started, logging into: /Users/prkumar/work/github/crc/test/e2e/out/test-results/e2e_2023-8-22_12-24-07.log
Running e2e test in: /Users/prkumar/work/github/crc/test/e2e/out/test-run
Working directory set to: /Users/prkumar/work/github/crc/test/e2e/out/test-run
Feature: Microshift test stories
Copying /Users/prkumar/work/github/crc/test/testdata/hello.yaml to /Users/prkumar/work/github/crc/test/e2e/out/test-run
Copying /Users/prkumar/work/github/crc/test/testdata/httpd-example-index.html to /Users/prkumar/work/github/crc/test/e2e/out/test-run
Copying /Users/prkumar/work/github/crc/test/testdata/httpd-example.yaml to /Users/prkumar/work/github/crc/test/e2e/out/test-run
Copying /Users/prkumar/work/github/crc/test/testdata/pipeline-sub.yaml to /Users/prkumar/work/github/crc/test/e2e/out/test-run
Copying /Users/prkumar/work/github/crc/test/testdata/tekton-task.yaml to /Users/prkumar/work/github/crc/test/e2e/out/test-run

  Background:
    Given setting config property "preset" to value "microshift" succeeds                                                                                      # testsuite.go:863 -> github.com/crc-org/crc/test/e2e/testsuite.SetConfigPropertyToValueSucceedsOrFails
    And setting config property "network-mode" to value "user" succeeds                                                                                        # testsuite.go:863 -> github.com/crc-org/crc/test/e2e/testsuite.SetConfigPropertyToValueSucceedsOrFails
    And executing single crc setup command succeeds                                                                                                            # testsuite.go:885 -> github.com/crc-org/crc/test/e2e/testsuite.ExecuteSingleCommandWithExpectedExitStatus
    And starting CRC with default bundle succeeds                                                                                                              # testsuite.go:751 -> github.com/crc-org/crc/test/e2e/testsuite.StartC
```

Even as part of e2e logs I can observe following
```
$ less  /Users/prkumar/work/github/crc/test/e2e/out/test-results/e2e_2023-8-22_12-24-07.log
[...]
12:24:07.785053  stdout: exitCodeOfLastCommandInShell=0
12:24:07.785203    bash: crc config set network-mode user
12:24:07.860932  stderr: level=debug msg="CRC version: 2.25.0+97705a\n"
12:24:07.860946  stderr: level=debug msg="OpenShift version: 4.13.6\n"
12:24:07.860950  stderr: level=debug msg="Podman version: 4.4.4\n"
12:24:07.860955  stderr: level=debug msg="Running 'crc config set'"
12:24:07.861209  stdout: Network mode changed. Please run `crc cleanup` and `crc setup`.
12:24:07.862019  stdout: exitCodeOfLastCommandInShell=0
[...]
```

**Comment by cfergeau** [comment]
> @adrianriobo I tested the scenario in mac and it does have `network-mode` setting. Is it only for windows?

`network-mode` is disabled in releases, but is available in builds from git.
https://github.com/crc-org/crc/blob/97705a008eda801ba291fe3b9982af464d4759e0/pkg/crc/config/settings.go#L106-L111

**Comment by adrianriobo** [comment]
I build released alike bits for testing + that should work also for pre-release testing

**Comment by praveenkumar** [comment]
> network-mode is disabled in releases, but is available in builds from git.

That's explain it, will work on that.

**Comment by praveenkumar** [comment]
fixed by #3798
